### PR TITLE
Remove Requires and implement a Makie extension instead

### PR DESCRIPTION
**Pull request by mohamed82008**
### Summary

- This PR implements Makie support as an extension instead of relying on Requires.jl.
- I also define the `visualize` function now at the top level in TopOpt.jl and overload it in the extension. This means that a single `visualize` function is overloaded for both continuum and truss problems. This function is also exported by default.
- This PR also avoids loading GLMakie in the extension and instead only loads and depends on Makie. Users can choose which backend they want to use themselves by loading GLMakie, WGLMakie or CairoMakie at the top level of their code.
- Plotting code is now tested using CairoMakie as a backend on CI which is easy to setup. This is just to avoid bit rotting of code.

### CODE REVIEW

**Comment by yijiangh** [comment]
From @mohamed82008 's description, the intention of this PR looks good to me. But I don't understand how `TopOptMakieExt` gets loaded. For example, in the beso script, I don't see you import `TopOptMakieExt`, how does Julia dispatch `visualize` to the right function?

**Comment by mohamed82008** [comment]
Extension packages in Julia get loaded automatically just like Requires worked in the past. So when I call `using Makie` and then `using TopOpt` (order doesn't matter), Julia will load the `TopOptMakieExt` module automatically. This module extends the otherwise empty `visualize` function with no methods by default. So if you try to use `visualize` without loading `Makie` first, you will get an error. A nicer experience is perhaps to define a fallback that throws an informative error asking the user to load `Makie`:

```julia
function visualize(arg::T; kwargs...) where {T}
    throw(MethodError("`visualize` is not defined for input type $T. This may be because the input to the function is incorrect or because you forgot to load `Makie` in your code. You can load `Makie` with `using Makie`."))
end
```

**Comment by mohamed82008** [comment]
Note that it is just a single function `TopOpt.visualize`. The extension module defines 2 extra methods for this function.

**Comment by mohamed82008** [comment]
The latest commit implements a more informative error message.

**Comment by yijiangh** [comment]
> Extension packages in Julia get loaded automatically just like Requires worked in the past. So when I call `using Makie` and then `using TopOpt` (order doesn't matter), Julia will load the `TopOptMakieExt` module automatically. This module extends the otherwise empty `visualize` function with no methods by default. So if you try to use `visualize` without loading `Makie` first, you will get an error. A nicer experience is perhaps to define a fallback that throws an informative error asking the user to load `Makie`:
> 
> ```julia
> function visualize(arg::T; kwargs...) where {T}
>     throw(ArgumentError("`visualize` is not defined for input type $T. This may be because the input to the function is incorrect or because you forgot to load `Makie` in your code. You can load `Makie` with `using Makie`."))
> end
> ```

Good to know! Do we have to declare extension packages somewhere, or we simply need to put it in to the `ext` folder?

File: Project.toml [hunk 1]
```diff
@@ -81,4 +79,10 @@ TimerOutputs = "0.5"
...
 WriteVTK = "1.9"
 Zygote = "0.6"
-julia = "1.5, 1.6"
+julia = "1.9"
+
+[weakdeps]
+Makie = "ee78f7c6-11fb-53f2-987a-cfe4a2b5a57a"
+
+[extensions]
+TopOptMakieExt = "Makie"
```
Comment (mohamed82008) [inline Project.toml:88]: @yijiangh these 2 new Project.toml sections are where we declare the extension packages

**Pull request merged**